### PR TITLE
Nodejs rework

### LINKED ISSUE #11: UI: Rework server portion of the UI to run on nodejs instead of python

### PR DESCRIPTION
**Pull request by Kristoffermg**
This PR includes the following changes:

1. Split all of the contents of index.html into multiple HTML and JS files. 
2. Removed the Python Flask implementation and replaced it with Node.js
3. Updated the README

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                integrity="sha256-20nQCchB9co0qIjJZRGuk2/Z9VM+kNiyxNV1lvTlZBo="
+                crossorigin=""></script>
+        <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH" crossorigin="anonymous">
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/flatpickr/dist/flatpickr.min.css">
+        <script src="https://cdn.jsdelivr.net/npm/flatpickr"></script>
+    </head>
+    <body>
+        <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+            <div class="container-fluid">
+            <a class="navbar-brand" href="#">Trace-Q</a>
```
Comment (KarmaKamikaze) [inline index.html]: nit: TRACE-Q er med stort :nerd_face: 

```suggestion
            <a class="navbar-brand" href="#">TRACE-Q</a>
```

File: index.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            </div>
+        </nav>
+
+        <div class="container mt-5">
+            <div class="row">
+                <div class="col-md-6 offset-md-3">
+                    <h1>Trace-Q trajectory simplifier</h1>
```
Comment (KarmaKamikaze) [inline index.html]: nit: TRACE-Q er med stort :nerd_face:

File: index.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Trace-Q</title>
```
Comment (KarmaKamikaze) [inline index.html]: nit: TRACE-Q er med stort :nerd_face: 
```suggestion
        <title>TRACE-Q</title>
```

**Review by Mailaur — APPROVED** [review]

File: markups/simplifyEndpoint.html [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        </div>
+    </div>
+    <input type="number" class="form-control" id="resolutionScale" value="2.0">
+</div>
+<div class="mb-2">
+    <label for="minRangeQueryAccuracy" class="form-label">Min query accuracy</label>
+    <button class="btn btn-light float-end mb-2" type="button" data-bs-toggle="collapse" data-bs-target="#minRangeQueryAccCollapse" aria-expanded="false" aria-controls="minRangeQueryAccCollapse">Info</button>
+    <div class="collapse" id="minRangeQueryAccCollapse">
+        <div class="card card-body">
+            The minimum query accuracy that the simplification method must uphold.
```
Comment (KarmaKamikaze) [inline markups/simplifyEndpoint.html]: Specificer at det er range query accuracy

File: markups/simplifyEndpoint.html [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        </div>
+    </div>
+    <input type="number" class="form-control" id="minRangeQueryAccuracy" value="0.95">
+</div>
+    <div class="mb-2">
+    <label for="minKnnQueryAccuracy" class="form-label">Min query accuracy</label>
+    <button class="btn btn-light float-end mb-2" type="button" data-bs-toggle="collapse" data-bs-target="#minKnnQueryAccCollapse" aria-expanded="false" aria-controls="minKnnQueryAccCollapse">Info</button>
+    <div class="collapse" id="minKnnQueryAccCollapse">
+        <div class="card card-body">
+            The minimum query accuracy that the simplification method must uphold.
```
Comment (KarmaKamikaze) [inline markups/simplifyEndpoint.html]: Specificer at det er KNN query accuracy

File: markups/simplifyEndpoint.html [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        </div>
+    </div>
+    <input type="number" class="dropdown form-control" id="rangeQueryGridDensity" value="0.1">
+</div>
+<div class="mb-2">
+    <label for="knnQueryGridDensity" class="form-label">KNN query grid density</label>
+    <button class="btn btn-light float-end mb-2" type="button" data-bs-toggle="collapse" data-bs-target="#knnQueryGridDenCollapse" aria-expanded="false" aria-controls="knnQueryGridDenCollapse">Info</button>
+    <div class="collapse" id="knnQueryGridDenCollapse">
+        <div class="card card-body">
+            Factor describing how close points appear in the grid for range queries.
```
Comment (KarmaKamikaze) [inline markups/simplifyEndpoint.html]:
```suggestion
            Factor describing how close points appear in the grid for KNN queries.
```

File: markups/simplifyEndpoint.html [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        </div>
+    </div>
+    <input type="number" class="form-control" id="rangeQueryTimeInterval" value="0.2">
+</div>
+<div class="mb-2">
+    <label for="knnQueryTimeInterval" class="form-label">KNN query time interval</label>
+    <button class="btn btn-light float-end mb-2" type="button" data-bs-toggle="collapse" data-bs-target="#knnQueryTimeIntervalCollapse" aria-expanded="false" aria-controls="knnQueryTimeIntervalCollapse">Info</button>
+    <div class="collapse" id="knnQueryTimeIntervalCollapse">
+        <div class="card card-body">
+            The factor that will be used to scale time intervals for each window in get_ids_from_knn queries.
```
Comment (KarmaKamikaze) [inline markups/simplifyEndpoint.html]:
```suggestion
            The factor that will be used to scale time intervals for each window in KNN queries.
```

**Review by KarmaKamikaze — CHANGES_REQUESTED** [review]
Lige lidt småting, men ellers helt fantastisk at se på! :heart_eyes:

File: markups/simplifyEndpoint.html [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+            Lower values will result in more queries.
+            Possible values: 0.1, 1
+            Lowest allowed KNN time interval multiplier is 0.02, otherwise number of query time points would be larger
+            than the allowed database connections.
```
Comment (Sarmisuper) [inline markups/simplifyEndpoint.html]: delete this

**Review by Sarmisuper — APPROVED** [review]

**Review by Flaiizor — APPROVED** [review]
I do not speak french, but this seems fine

**Pull request merged**